src/main.c

```diff
@@ -46,23 +46,29 @@ static char *stdin_slurp(int *sz) {
 		/* this is faster but have limits */
 		/* run test/add10k.sh script to benchmark */
 		const int buf_size = 96096;
-		buf = calloc (1, buf_size);
-		if (!fgets (buf, buf_size - 1, stdin)) {
+
+		buf = malloc (buf_size);
+		if (!fgets (buf, buf_size, stdin)) {
 			free (buf);
 			return NULL;
 		}
 		if (feof (stdin)) {
 			free (buf);
 			return NULL;
 		}
-		int buf_len = strlen (buf) - 1;
-		buf[buf_len] = 0;
-		char *newbuf = realloc (buf, buf_len + 1);
-		if (newbuf) {
-			return newbuf;
+
+		size_t buf_len = strlen (buf);
+		if (buf_len > 0) {
+			buf[buf_len - 1] = '\0';
 		}
-		free (buf);
-		return NULL;
+
+		char *newbuf = realloc (buf, buf_len);
+		// realloc behaves like free if buf_len is 0
+		if (!newbuf && buf_len > 0) {
```

Comment:
this check is wrong. if allocation fails it will leak if the new buflen is <1 because it will return null, but buf will not be freed